java/spark/v3/src/main/java/com/lancedb/lance/spark/LanceDataSource.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.spark;
+
+import com.lancedb.lance.spark.internal.LanceConfig;
+import com.lancedb.lance.spark.internal.LanceReader;
+import org.apache.spark.sql.connector.catalog.Table;
+import org.apache.spark.sql.connector.catalog.TableProvider;
+import org.apache.spark.sql.connector.expressions.Transform;
+import org.apache.spark.sql.sources.DataSourceRegister;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+import java.util.Map;
+
+public class LanceDataSource implements TableProvider, DataSourceRegister {
+
+  @Override
+  public StructType inferSchema(CaseInsensitiveStringMap options) {
```

Comment:
are we actually inferring schema here?

Reply:
Good catch

Reply:
Double checked that inferSchema is to get the schema of a given table given the table info passed in via
```
    data = spark.read().format("lance")
        .option("db", dbPath)
        .option("table", TestUtils.TestTable1Config.tableName)
        .load();
```
It will not and does not contains another "schema" (e.g. schema with selected columns).
Validated it by looking at pinot and other spark connectors